crypto/mem_dbg.c

```diff
@@ -112,7 +112,8 @@ int CRYPTO_mem_ctrl(int mode)
 #else
     int ret = mh_mode;
 
-    CRYPTO_THREAD_run_once(&memdbg_init, do_memdbg_init);
+    if (!RUN_ONCE(&memdbg_init, do_memdbg_init))
+        return 0;
```

Comment:
Maybe it should be made to return -1, just like `CRYPTO_mem_leaks`.  What ddo you think?